### ISSUE TITLE
Segfault on start (or shortly after)

### ISSUE THREAD

**Issue by liquidonline** [psifidotos/Latte-Dock#319]
Hello,

I recently did a fresh install of KDE neon, and with it a fresh pull of latte-dock.  The previous one is probably close to a months old, but it worked fine on the old installed OS.   Upon start the first time, it worked well.  I made a few tweaks to config (can't remember which, I wasn't paying that much attention and I don't know where config is stored).   Since then, I get constant segfaults on start.

The old one also segfaults.  I suspect there is something off on my end that changed with the new OS install, but I don't know what.

If you need any other data, let me know.  In the interim, I'd like to know how I can clear the configuration and validate if any of the specific settings I changed can be the cause of this.

OS: KDE Neon 5.9.4 user edition
Kernel: 4.8.0-41-generic 64bit
Plasma 5.9.4
QT 5.7.1 

DDD output:
```
GNU DDD 3.3.12 (x86_64-pc-linux-gnu), by Dorothea LReading symbols from latte-dock...(no debugging symbols found)...done.
(gdb) file /usr/bin/latte-dock
Reading symbols from /usr/bin/latte-dock...(no debugging symbols found)...done.
(gdb) core /home/sandro/.ddd/sessions/.ddd/dddcore
[New LWP 13433]
[New LWP 13437]
[New LWP 13439]
[New LWP 13438]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/bin/latte-dock'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007ffff407d809 in QWindow::winId() const () from /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5
[Current thread is 1 (Thread 0x7fffe5f768c0 (LWP 13433))]
(gdb) run
Starting program: /usr/bin/latte-dock 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe3a6c700 (LWP 13451)]
[New Thread 0x7fffe1f52700 (LWP 13452)]
[New Thread 0x7fffdbfff700 (LWP 13453)]

Thread 1 "latte-dock" received signal SIGSEGV, Segmentation fault.
0x00007ffff407d809 in QWindow::winId() const () from /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5
(gdb) frame 9
```
Backtrace:
![image](https://cloud.githubusercontent.com/assets/4250785/24280320/2680f8dc-1026-11e7-91c9-dbf48f64be46.png)

**Comment by audoban** [comment]
@psifidotos The backtrace is bit useless :(, but it seems to happen in libtaskmager.

**Comment by psifidotos** [comment]
Constant crashes on startup on my system happened when i had enabled both previews and hovering effect... you could try to delete at .config, the files:

latte-dock-appletsrc
latte-dockrc

**Comment by psifidotos** [comment]
#268 the user had the same issue...

**Comment by psifidotos** [comment]
If plasma 5.9.4 introduced something new causing this then when it will land at opensuse it should occur also in my system....

**Comment by audoban** [comment]
@psifidotos Confirm me when you can, I will follow in plasma 5.8 for now.

**Comment by liquidonline** [comment]
My work laptop is on plasma 5.9.4 and same qt version 5.7.1.  However, the Latte-Dock build here works still.  I'll need to verify the config as you suggested, so perhaps it's not as simple as just a change in plasma 5.9.4?  I'll fill you guys in this weekend :)

I didn't check files to see if a build number is present, but the last commit from git log shows fix for #215, Mar 16 21:44:16 2017 +0200.  Sorry for all the "I don't know" comments, but I'm in the key part of the year at my work, so my focus is elsewhere to say the least, but I figured at least a quick logging of the problem would be more helpful than swearing at it ;)

**Comment by psifidotos** [comment]
@audoban plasma 5.9.4 works just fine in my system... I had crashes though because of the hovering effect #323 for which I added a workaround and reported upstream the issue...

@liquidonline can you check please if the latest master fixes this situation for you ?

**Comment by psifidotos** [comment]
@liquidonline I take that this was related to the hovered effect... if it happens again let us know...

**Comment by liquidonline** [comment]
Hi,
I know this is closed now, but I wanted to circle back to validate/document for anyone else.
You're right about the hover effect.  The Highlight on hovering effect that was causing the segfault.

I've  just pulled the latest master and I can also confirm that whatever you fixed in commit 1156a55626b68a41de640207126b965bf96c5a09 for this bug has resolved it on my system.

Great to see such active development on this project.  Kudos to all.
